timdex_dataset_api/dataset.py

```diff
@@ -264,7 +184,7 @@ def _get_filtered_dataset(
 
         return self.dataset.filter(combined_expressions)
 
-    def _parse_date_filters(self, run_date: str | date | None) -> dict:
+    def _parse_date_filters(self, run_date: str | date | None) -> DatasetFilters:
```

Comment:
This function signature caught my eye when I saw the return type was `DatasetFilters`. This is essentially saying it returns a `TypedDict`, right? And because `total=False`, the type checker doesn't throw an error even though the function returns dictionary with only a _subset_ of the keys defined in `DatasetFilters`?

Reply:
That's right!  It tells you that it returns a valid `DatasetFilters` object, but it need not be complete.